amp/WEB-INF/src/org/dgfoundation/amp/onepager/components/AmpFundingAmountComponent.java

```diff
@@ -139,28 +144,61 @@ public IConverter getInternalConverter(java.lang.Class<?> type) {
             amount.getTextContainer().add(new AttributeModifier("size", new Model<String>("9")));
         else
             amount.getTextContainer().add(new AttributeModifier("size", new Model<String>(amountSize)));
-        
+
         add(amount);
-        
-        
+
+
+        PropertyModel<AmpCurrency> currentCurrency = new PropertyModel<AmpCurrency>(model, propertyCurrency);
+        List<AmpCurrency> activeCurrencies = CurrencyUtil.getUsableNonVirtualCurrencies();
+        boolean isCurrencyPresentInList = activeCurrencies.stream()
+                .anyMatch(curr -> curr.getAmpCurrencyId().equals(currentCurrency.getObject().getAmpCurrencyId()));
+
+        if (!isCurrencyPresentInList) {
+            activeCurrencies.add(currentCurrency.getObject());
+            activeCurrencies.sort(Comparator.comparing(AmpCurrency::getCurrencyCode));
+        }
+
         AbstractReadOnlyModel<List<AmpCurrency>> currencyList = new AbstractReadOnlyModel<List<AmpCurrency>>() {
             @Override
             public List<AmpCurrency> getObject() {
-                return (List<AmpCurrency>) CurrencyUtil.getUsableNonVirtualCurrencies() ;
+                return activeCurrencies;
             }
         };
-        
-        currency = new AmpSelectFieldPanel<AmpCurrency>("currency",
+
+        currency = new AmpStyledSelectFieldPanel<AmpCurrency>("currency",
                 new PropertyModel<AmpCurrency>(model, propertyCurrency),
-                currencyList, fmCurrency, hideLabel, false, null, hideNewLine) {
+                currencyList, fmCurrency, hideLabel, false, new IOptionRenderer<AmpCurrency>() {
```

Comment:
The code is not compiling, please notice that you have an extra false after hideLabel and that false is being taken by renderer in the AmpStyledSelectFieldPanel convertor. Ill continue validating but please fix so it compiles